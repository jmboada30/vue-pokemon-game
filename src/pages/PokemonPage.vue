<template>
  <h1 v-if="!pokemon">Por favor espere...</h1>

  <div v-else>
    <h1>Quien es este Pokemon?</h1>
    <pokemon-picture
      :pokemonId="pokemon.id"
      :showPokemon="showPokemon"
    ></pokemon-picture>
    <pokemon-options :pokemons="pokemonArr" @selection="checkAnswer"></pokemon-options>
  </div>
</template>

<script>
import PokemonPicture from "@/components/PokemonPicture.vue";
import PokemonOptions from "@/components/PokemonOptions.vue";
import getPokemonOptions from "@/helpers/getPokemonOptions";
export default {
  components: { PokemonPicture, PokemonOptions },
  data() {
    return {
      pokemonArr: [],
      pokemon: null,
      showPokemon: false,
    };
  },
  methods: {
    async mixPokemonArray() {
      this.pokemonArr = await getPokemonOptions();
      this.selectPokemon();
    },
    selectPokemon() {
      const index = Math.floor(Math.random() * 4);
      this.pokemon = this.pokemonArr[index];
    },
    checkAnswer(pokemonId) {
      this.showPokemon = true;
      if(pokemonId === this.pokemon.id) {
        alert('YES')
      } else {
        alert('NOO')
      }
    }
  },
  mounted() {
    this.mixPokemonArray();
  },
};
</script>

<style></style>
