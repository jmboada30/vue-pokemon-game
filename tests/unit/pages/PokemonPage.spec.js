import { mount, shallowMount } from '@vue/test-utils';
import PokemonPage from '@/pages/PokemonPage';
import { pokemonArr } from '../mocks/pokemons.mock';
describe('PokemonPage Page', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(PokemonPage);
  });

  test('debe hacer match con el snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('debe llamar mixPokemonArray al montar', () => {
    const mixPokemonArray = jest.spyOn(PokemonPage.methods, 'mixPokemonArray');
    const wrapper = shallowMount(PokemonPage);
    expect(mixPokemonArray).toHaveBeenCalled();
  });

  test('debe hacer match con el snapshot cuando cargan los pokemons', () => {
    const wrapper = mount(PokemonPage, {
      data() {
        return {
          pokemonArr: pokemonArr,
          pokemon: pokemonArr[0],
          showPokemon: false,
          showAnswer: false,
          message: '',
        };
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  test('debe mostrar los components de PokemonPicture y PokemonOptions', () => {
    const wrapper = shallowMount(PokemonPage, {
      data() {
        return {
          pokemonArr: pokemonArr,
          pokemon: pokemonArr[0],
          showPokemon: false,
          showAnswer: false,
          message: '',
        };
      },
    });

    const pokemonPicture = wrapper.find('pokemon-picture-stub');
    const pokemonOptions = wrapper.find('pokemon-options-stub');
    console.log(pokemonOptions.attributes());

    expect(pokemonPicture.exists()).toBe(true);
    expect(pokemonOptions.exists()).toBe(true);
    expect(pokemonPicture.attributes('pokemonid')).toBe('1');
    expect(pokemonOptions.attributes('pokemons')).toBeDefined();
  });
});
