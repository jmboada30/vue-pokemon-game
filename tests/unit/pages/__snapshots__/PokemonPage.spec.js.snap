// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage Page debe hacer match con el snapshot 1`] = `<h1>Por favor espere...</h1>`;

exports[`PokemonPage Page debe hacer match con el snapshot cuando cargan los pokemons 1`] = `
<div>
  <h1>Quien es este Pokemon?</h1>
  <div class="pokemon-container"><img class="hidden-pokemon" src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" alt="pokemon"></div>
  <div class="options-container">
    <ul>
      <li>bulbasaur</li>
      <li>ivysaur</li>
      <li>venusaur</li>
      <li>charmander</li>
    </ul>
  </div>
  <!--v-if-->
</div>
`;
